### DISCUSSION TITLE
There is a major bug in the exalidraw plugin in Trilium, if Ctrl+Z, the page will be empty or switch to an unrelated page.

### DISCUSSION THREAD

**Discussion by tiger7651** [TriliumNext/Trilium#4437] [General]
The excalidraw plugin has a major bug, if Ctrl+Z, the page will be empty or switch to an unrelated page.

**Comment by zadam** [comment]
Hi, what platform is that? Browser or desktop?

For me CTRL-Z seems to work just fine.

**Reply by tiger7651** [reply]
The excalidraw plugin in Trilium,

This bug will not appear immediately,

When operating multiple pages and then Ctrl+Z, the page will disappear and all previous work will be wasted

**Reply by zerebos** [reply]
> The excalidraw plugin in Trilium,

Do you mean the desktop app then? Or is this the web version?

Maybe a demo or more concrete steps to recreate could help here.

**Reply by tiger7651** [reply]
On desktop applications
It's like this:
When you create an excalidraw canvas note with an image, and then create another canvas note with an image, in the second note, press Ctrl+Z, and the page will be erased or switched to the page of the first note.
This bug often occurs and is easy to reproduce, as it can occur in any canvas note that contains images.
In the new version v.0.61.14, it still exists.

![Video_2023-11-17_150812](https://github.com/zadam/trilium/assets/150779117/e704cbf3-fce8-4cc0-ab34-b4cbd89c259c)

**Comment by rodrigoalexandrecosta** [comment]
I can confirm this behavior. I'm using the desktop app in Fedora KDE, v.0.61.12.

I was testing this yesterday and was able to reproduce what @tiger7651 reported. I just added a text to a Canvas note, un-selected it and hit `Ctrl+Z`. The entire Canvas was erased, although I could redo with `Ctrl+Shift+Z`. 

Today I was also trying to reproduce the bug, and I don'tk now if was because I was editing another Canvas note just before that, but the actual content of the Canvas that I was using to reproduce the bug was erased and overwrote by the content of the Canvas note that I was editing before that. In this case, undo/redo didn't work and I had to restore a revision (tip to @tiger7651 - you can restore a note to previous state by using the "Note Revisions" button in the ribbon menu at the top-right of the note).

I'm trying to record my screen to post a gif, but isn't working, I have some kind of issue in my Wayland session that's not allowing me to record it. I'll update this if I can solve it.

**Comment by zadam** [comment]
Hi, thanks all helping to reproduce the problem - the issue was that we're re-using the same excalidraw instance when going from one note to another (for performance reasons) and the edit history sort of leaked from one note to another. This issue should be resolved in the next version.
